src/cases/containers/case-hearings/case-hearings.component.ts

```diff
@@ -108,23 +103,26 @@ export class CaseHearingsComponent implements OnInit, OnDestroy {
     this.upcomingHearings$ = this.getHearingListByStatus(EXUISectionStatusEnum.UPCOMING);
     this.pastAndCancelledHearings$ = this.getHearingListByStatus(EXUISectionStatusEnum.PAST_OR_CANCELLED);
     this.listedHearings$ = this.getHearingListByStatus(EXUIDisplayStatusEnum.LISTED);
-
-    this.roleCatSubscription = this.roleCategoryMappingService.getUserRoleCategory(this.userRoles$).subscribe(
-      (userRole) => {
-        this.hearingsActions = [Actions.READ];
-        if (userRole === UserRole.LegalOps) {
-          this.hearingsActions = [...this.hearingsActions, Actions.CREATE, Actions.UPDATE, Actions.DELETE];
-        } else if (userRole === UserRole.Ogd) {
-          this.isOgdRole = true;
-        }
-        if (this.hearingsActions.includes(Actions.CREATE)) {
-          this.hasRequestAction = true;
-        }
-        if (this.hearingsActions.includes(Actions.CREATE)) {
-          this.hasRequestAction = true;
-        }
+    this.userRoles = [];
+    const detailsStr = this.sessionSvc.getItem('userDetails');
+    if (detailsStr) {
+      const details = JSON.parse(detailsStr) as object;
+      if (details && details.hasOwnProperty('roles')) {
```

Comment:
@johnbenjamin-hmcts - can be rewritten with optional chaining, i.e. `details?.hasOwnProperty('roles')`